source/simple_sub_pub_demo.c

```diff
@@ -422,7 +422,7 @@ static bool prvSubscribeToTopic( MQTTQoS_t xQoS,
 }
 
 /*-----------------------------------------------------------*/
-
+volatile uint32_t ulQoS0FailureCount = 0UL, ulQoS1FailureCount = 0UL;
```

Comment:
I'm not sure if just declaring these variables as volatile is enough, since it looks like they could be incremented from multiple tasks concurrently? Unless we expect this to be a very rare occurrence and so don't care too much about exactness or data races.

Reply:
Right, don't really care about those data races as the values are not critical as far as seeing that a count has incremented.  A data race could conceivably result in one increment being missed, but only if there are two at once, so the value will still increment just by not enough.